### PR TITLE
feat: refactor main scenes to store, add video on main page

### PR DESCRIPTION
**Pull request by DzianisPasiukou**

### CODE REVIEW

File: apps/js-machine-front/src/app/stores/ui.store.ts [hunk 1]
```diff
@@ -1 +1,13 @@
...
+import { RootStore } from "@js-machine-app/front/stores/root.store";
+
+export class UiStore {
+  @observable public isLoading?: boolean;
+
+  public constructor(private rootStore: RootStore) {
+  }
+
+  @action public setIsLoading =  (isLoading: boolean) => {
+    runInAction(() => (this.isLoading = isLoading));
```
Comment (vharadkou) [inline apps/js-machine-front/src/app/stores/ui.store.ts]: you don't need runInAction if your method is synchronous and it marked as @action

File: apps/js-machine-front/src/app/stores/ui.store.ts [hunk 2]
```diff
@@ -1 +1,13 @@
-export class UiStore {}
+import { action, observable, runInAction } from "mobx";
+import { RootStore } from "@js-machine-app/front/stores/root.store";
+
+export class UiStore {
+  @observable public isLoading?: boolean;
```
Comment (vharadkou) [inline apps/js-machine-front/src/app/stores/ui.store.ts]: Think that solution around single flag for all async calls is not the best option

**Review by vharadkou — CHANGES_REQUESTED** [review]

File: apps/js-machine-front/src/app/stores/ui.store.ts (hunk 2 above)
Comment (DzianisPasiukou) [reply apps/js-machine-front/src/app/stores/ui.store.ts]: I suppose it's like isPageLoading, if separate loading flags is needed, we'll add that

File: apps/js-machine-front/src/app/stores/ui.store.ts (hunk 1 above)
Comment (DzianisPasiukou) [reply apps/js-machine-front/src/app/stores/ui.store.ts]: fixed, thanks

**Review by vharadkou — APPROVED** [review]

**Pull request merged**